tools/ci-build/publisher/src/subcommand/upgrade_runtime_crates_version.rs

```diff
@@ -46,11 +84,9 @@ pub async fn subcommand_upgrade_runtime_crates_version(
 
 fn update_gradle_properties<'a>(
     gradle_properties: &'a str,
-    upgraded_version: &'a semver::Version,
+    upgraded_version: &semver::Version,
+    version_regex: &Regex,
```

Comment:
Nit: Instead of accepting a regex from the caller, it may be safer to determine which regex should be used locally, thus minimizing the risk of an incorrect regex being passed in. I suggest introducing an `enum` with two variants, each containing a `semver::Version` as a parameter. Within the function, you can match on these variants to select the appropriate regex. Although the regex should remain static, it shouldn't be passed in by the caller.

Reply:
I like the suggestion. Conveniently, we already have an enum `PackageStability` in `smithy_rs_tool_common::package`. The function can take that enum instead of a regex. Updated in 411b21d.